### ISSUE TITLE
Introduce session setting for applied timezone.

### ISSUE THREAD

**Issue by f1yegor** [ClickHouse/ClickHouse#34600]
**Use case**

I want my reports to show data to customers in different timezones, for example histogram of daily visits. And not manually and tediously make the conversion for a particular timezone. One session setting like "applied_time_zone" should be used to convert all dates/datetimes in queried data.
Functions like `toStartOfDay`, `toStartOfWeek` should return correct time in applied timezone.

Now we have parameter in clickhouse-client: 
```
--use_client_time_zone arg

Use client timezone for interpreting 
DateTime string values, instead of 
adopting server timezone.
```

The proposed setting should work on a session level `WITH SETTINGS ....` and dynamically applied.

**Describe alternatives you've considered**

There are several ways to workaround this: 
1. manually convert each date/datetime value `toDateTime( , timezone)`
2. use `--use_client_time_zone`, but only applicable to client, and not language libraries.

**Comment by Enmk** [comment]
@zvonand works on this

**Comment by zvonand** [comment]
closed by recently merged #44149
